### ISSUE TITLE
Docstrings on functions matching "no-docstring-rgx" are not linted

### ISSUE THREAD

**Issue by dkbarn** [pylint-dev/pylint#9215]
### Bug description

```python
When no-docstring-rgx is defined and allows docstrings to be optional on certain functions, those optional docstrings which are present are not being linted.

Consider the following private function as an example:

def _private_fn(arg):
    """This docstring is not required but I've still chosen to write it.

    Args:
        argh: Oops this argument has a typo in the name.
    """
```
```


### Configuration

```ini
[BASIC]

# Regular expression which should only match function or class names that do
# not require a docstring.
no-docstring-rgx=^_
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
Nothing
```

### Expected behavior

I would expect to see this output:
```
pylint:W9015:"arg" missing in parameter documentation
pylint:W9017:"argh" differing in parameter documentation
```

My expectation is that "no-docstring-rgx" simply defines when docstrings are optional, it does not mean linting is completely disabled on docstrings that are present.

### Pylint version

```shell
pylint 3.0.2
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. Your interpretation of the docs here is the most straightfoward, but it looks like the way this option has (always?) functioned is as a `skip-docstrings-rgx`, as it currently behaves. We should decide whether that's worth breaking or worth documenting better instead.

**Comment by Pierre-Sassoulas** [comment]
How about creating a new option named correctly  (doing the same thing) and keeping the old one until we have a config upgrader ? (But de upgrade pylint to use the new name)
